user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/tree/SolutionTree.java

```diff
@@ -579,17 +565,42 @@ public void setShowLocalizedFileNames( boolean showLocalizedFileNames ) {
     this.showLocalizedFileNames = showLocalizedFileNames;
     // use existing tree and switch text/title
     for ( int i = 0; i < getItemCount(); i++ ) {
-      toggleLocalizedFileNames( (FileTreeItem) getItem( i ) );
+      refreshTreeItemDisplay( (FileTreeItem) getItem( i ) );
```

Comment:
Couldn't we call this 'for' block only when the new and the old values differed? But maybe it's safer this way...